pkg/mixin/provider/delete.go

```diff
@@ -0,0 +1,39 @@
+package mixinprovider
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/deislabs/porter/pkg/mixin"
+	"github.com/pkg/errors"
+)
+
+func (fs *FileSystem) Delete(opts mixin.DeleteOptions) (*mixin.Metadata, error) {
+	if opts.Name != "" {
+		return fs.deleteByName(opts.Name)
+	}
+
+	return nil, errors.New("No mixin name was provided for deletion")
+}
+
+func (fs *FileSystem) deleteByName(name string) (*mixin.Metadata, error) {
+	mixinsDir, err := fs.GetMixinsDir()
+	if err != nil {
+		return nil, err
+	}
+	mixinDir := filepath.Join(mixinsDir, name)
+	exists, _ := fs.FileSystem.Exists(mixinDir)
+	if exists == true {
+		err = fs.FileSystem.RemoveAll(mixinDir)
+		if err != nil {
+			return nil, err
+		}
+
+		m := mixin.Metadata{
+			Name: name,
+		}
+		return &m, nil
+	}
+
+	return nil, errors.New(fmt.Sprintf("Could not find %s in the mixin directory.", name))
```

Comment:
So in this case, where they asked to delete a mixin that isn't installed, instead of having that be an error, let's instead print a debug message that the mixin was not installed and allow the command to succeed. 

Here is the [doc](https://github.com/deislabs/porter/blob/master/CONTRIBUTING.md#logging) how to do debug logging. You can tell if we are in debug mode by checking `fs.Debug`.